### PR TITLE
Add Experimental Support for Gemma variants [1B, 27B]

### PR DESCRIPTION
**Pull request by MohammedTaherMcW**
### Ticket
[Link to JIRA Issue
](https://flexaihq.atlassian.net/browse/FMIOT-1?atlOrigin=eyJpIjoiZjc3N2NiYjkzMGQwNDJjMDhkODZkNTUzM2M0MmE5YmMiLCJwIjoiaiJ9)
### Problem description
Enable Support for Gemma-3-27b-it Model.

### What's changed
Added support for gemma-3-27b-it model
Updated model_config.py to support gemma-3-27b-it , including end-of-sequence (EoS) token handling.
Updated load_checkpoints.py to support gemma-3-27b-it weight loading.
Modified apply_scaling logic to handle both LLaMA and gemma-3-27b-it model.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/tg-quick-trigger.yaml) CI passes (if applicable)
- [ ] [TG demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/tg-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: models/tt_transformers/tt/model.py [hunk 1]
```diff
@@ -188,7 +188,14 @@ def prepare_inputs_prefill(self, tokens, start_pos=0, page_table=None, chunk_pag
...
             tt_chunk_page_table = None
 
-        return tokens_embd, tt_rot_mats_prefill_global, tt_rot_mats_prefill_local, tt_page_table, tt_chunk_page_table
+        return (
+            tokens_embd,
+            tt_rot_mats_prefill_global,
+            tt_rot_mats_prefill_local,
+            tt_page_table,
+            tt_chunk_page_table,
+            None,
```
Comment (jschuhmacher) [inline models/tt_transformers/tt/model.py:197]: Why does `prepare_inputs_prefill` return an extra None here?

File: models/tt_transformers/tt/model.py [hunk 2]
```diff
@@ -317,6 +324,7 @@ def ttnn_prefill_forward(
         chunk_start_idx=None,
         get_last_token=-1,
         kv_cache=None,
+        **kwargs,
```
Comment (jschuhmacher) [inline models/tt_transformers/tt/model.py:327]: This makes it hard to see which arguments could be expected. Why not list the potential arguments with default values above?

File: models/tt_transformers/tt/model.py [hunk 3]
```diff
@@ -361,6 +369,7 @@ def ttnn_decode_forward(
         page_table=None,
         kv_cache=None,
         argmax_on_device=False,
+        **kwargs,
```
Comment (jschuhmacher) [inline models/tt_transformers/tt/model.py:372]: This makes it hard to see which arguments could be expected. Why not list the potential arguments with default values above?

File: models/experimental/gemma3/tests/vision_tests/test_mmp.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
-"""Gemma-3-4b-it Test for multi-modal-projector"""
+"""Gemma3 Test for multi-modal-projector"""
 
 
-# SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+# SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
```
Comment (jschuhmacher) [inline models/experimental/gemma3/tests/vision_tests/test_mmp.py:4]: General comment for all the files: it seems Tenstorrent prefers the 
```
# SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
```
text instead.

File: models/experimental/gemma3/tests/test_decoder.py [hunk 5]
```diff
@@ -137,12 +133,21 @@ def test_decoder_inference(
         layer_num=0,
         weight_cache_path=model_args.weight_cache_path(dtype),
         transformation_mats=transformation_mats,
-        transformation_mats_local=transformation_mats_local,
         paged_attention_config=paged_attention_config,
     )
 
     seqlen = 1
 
+    cos, sin = precompute_freqs(
```
Comment (jschuhmacher) [inline models/experimental/gemma3/tests/test_decoder.py:141]: Same as above, the rope setup and layer seem to be mismatching

File: models/experimental/gemma3/tests/test_attention.py [hunk 6]
```diff
@@ -97,6 +101,7 @@ def test_attention_inference(
         model_args.rope_theta,
         model_args.rope_scaling,
     )
+
```
[the remark this answers is not in the document]
Comment (jschuhmacher) [reply models/experimental/gemma3/tests/test_attention.py:104]: Maybe another test with a global attention layer is also interesting?

File: models/experimental/gemma3/tt/text_model.py [hunk 7]
```diff
@@ -452,6 +588,15 @@ def forward(
             elif activation_dtype is not None and x.dtype != activation_dtype:
                 x = ttnn.typecast(x, activation_dtype)
```
Comment (jschuhmacher) [inline models/experimental/gemma3/tt/text_model.py:589]: Nitpick, naming the masks first makes it a bit easier to read, for instance:
```suggestion
            causal_mask = None
            if attention_masks is not None:
                sliding_causal_mask, global_causal_mask = attention_masks
                causal_mask = sliding_causal_mask if getattr(layer.attention, "is_sliding", False) else global_causal_mask
```
